design/specs/navigation.md

```diff
@@ -0,0 +1,133 @@
+
+### Design Goals
+
+- Users clearly understand which page they're on
+- Users easily navigate to other pages
+- Users may anticipate the result of navigation interactions
+
+## Navigation structure diagram
+
+The navigation design will be based on the following structure:
+
+| 1st Level         | 2nd Level       | 3rd Level     | 4th Level   |
+|-------------------|-----------------|---------------|-------------|
+| Database Page     | Schema Page     | Table Page    | Record Page |
+|                   | Data Explorer   | Query Page    |             |
+|                   | Data Import     |               |             |
+
+- 1st level: Database
+- 2nd level: Schema
+- 3rd level: Entities
+- 4th level: Record
+
+[Navigation by Page](https://balsamiq.cloud/sjfrgln/pbe3tnu/r7FA3?f=N4IgUiBcCMA0IDkpxAYWfAMhkAhHAsjgFo4DSUA2gLoC%2BQA%3D)
+
+## Phasing Out Tab Navigation
+
+Tabs will be phased out as part of the new navigation improvements.
+
+## Navigation Methods
+
+Navigation within Mathesar will take place in three different ways:
+
+- Through the site header
+- Through links in pages
+- Direct access through page URLs
+
+### Navigation via site header
+
+- The logo in the site header will lead visitors to the site's root, in this case the database homepage.
+- A breadcrumb-like navigation will be displayed at the top part of the site's header as the user navigates through the pages.
+- When users click on a breadcrumb element, a [page-selector](#page-selector) component is displayed with a list of schemas or entities based on the breadcrumb level.
+- Upon reaching the record level, the top navigation will show a 'Go to Record' action, which will open the [Record Selector](/engineering/specs/record-selector) component.
+- The current page will not be included in the breadcrumb path.
+- The current page label will be placed underneath the top header component in a sub-header component.
+- An icon will be used to identify each type of page (table, schema, record etc.)
+- When we eventually add support for several databases, we'll need CRUD UI for databases, and we may design some additional UI that stays within the header at that point.
+
+### Navigation via links within pages
+
+Throughout Mathesar, the following links exist within pages which allow for more targeted navigation to other pages.
+
+- **Database Page**
+  - Links to **Schema Pages** in a list of schemas.
+
+- **Schema Page**
+  - Links to **Table Pages** in a list of tables.
+  - Links to **Query Pages** in a list of queries.
+
+- **Table Page**
+  - Within the table data:
+    - Links to the **Record Page** for each row via the row's primary key cell value. _Note: Links within cell values will only be clickable when the cell is selected. This UX should be consistent with the behavior of the "URL" UI Type._
+    - Links to **Record Pages** for related records via the FK cell values
```

Comment:
Could we also link to the table page for the related table from the FK column header or something like that?

I think it would be nice for the user to be able to navigate to a related table without opening up the table inspector, but maybe that's over-optimizing.

Reply:
> link to the table page for the related table from the FK column header

I'm hesitant about going this direction for the following reasons:

- In preliminary conversations about the Table Inspector design, I think Ghislaine and I have been coalescing on the idea that _clicking a column header should select all the cells within that column_. It's hard for me to imagine how to preserve that cell-selection design while also allowing the user to navigate to a Table Page via the column header.
- I'd like to stick to the design we [established in Matrix on Tuesday](https://matrix.to/#/!xInuTkBwjXZXYatIlm:matrix.mathesar.org/$JIGcTbMy-KCHeG5QAN0RkMa6fPT7npl32JdgVkbvIeI?via=matrix.mathesar.org&via=matrix.org), which keeps the column header as simple as possible. FK column get a link icon but nothing else special.

> or something like that

I'm open to hearing other ideas more specifically

Reply:
Let's not worry about this. If we run into use cases where navigating between tables needs to be made easier, we can figure it out then.

Reply:
> In preliminary conversations about the Table Inspector design, I think Ghislaine and I have been coalescing on the idea that clicking a column header should select all the cells within that column. 

This is consistent with the selection patterns I'm using for the table inspector UX. I think it should remain simple.